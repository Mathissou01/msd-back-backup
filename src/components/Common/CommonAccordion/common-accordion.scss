@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/02-tools/tools.pictos" as p;

.c-CommonAccordion {
  width: 100%;
  min-height: 70px;
  background-color: colors.$white;
  border-top: 1px solid colors.$blue-4;

  &:last-of-type {
    border-bottom: 1px solid colors.$blue-4;
  }

  &__ContentRow {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 100%;
    min-height: 70px;
    padding: 12px;

    &_content {
      width: 85%;
    }

    &_expander {
      display: flex;
      align-items: center;
      cursor: pointer;

      &_label {
        margin-right: 12px;
      }

      &_icon {
        @include p.get-picto("collapseArrow");

        display: block;
        width: 16px;
        height: 16px;
        transform: rotate(-90deg);
        transform: rotate(-90deg);
      }

      &_iconExpanded {
        transform: rotate(0);
        transform: rotate(0);
      }
    }
  }

  &__ExpandedContent {
    display: flex;
    align-items: center;
    height: 100%;
    min-height: 70px;
    padding: 12px;
  }
}
