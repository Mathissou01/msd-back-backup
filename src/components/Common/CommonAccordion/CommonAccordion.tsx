import { useState } from "react";
import classNames from "classnames";
import "./common-accordion.scss";

interface ICommonAccordionProps {
  content: JSX.Element;
  expandedContent: JSX.Element;
  expanderLabel?: string;
  expanderNotAvailable?: boolean;
}

export default function CommonAccordion({
  content,
  expandedContent,
  expanderLabel,
  expanderNotAvailable,
}: ICommonAccordionProps) {
  const [isExpanded, setIsExpanded] = useState<boolean>(false);
  return (
    <div className="c-CommonAccordion">
      <div className="c-CommonAccordion__ContentRow">
        <div
          className="c-CommonAccordion__ContentRow_content"
          data-testid="accordion-content"
        >
          {content}
        </div>
        {!expanderNotAvailable && (
          <div
            className="c-CommonAccordion__ContentRow_expander"
            onClick={() => setIsExpanded(!isExpanded)}
            data-testid="accordion-expander"
          >
            {expanderLabel && (
              <span className="c-CommonAccordion__ContentRow_expander_label">
                {expanderLabel}
              </span>
            )}
            <span
              className={classNames(
                "c-CommonAccordion__ContentRow_expander_icon",
                {
                  ["c-CommonAccordion__ContentRow_expander_iconExpanded"]:
                    isExpanded,
                },
              )}
            />
          </div>
        )}
      </div>
      {!expanderNotAvailable && isExpanded && (
        <div
          className="c-CommonAccordion__ExpandedContent"
          data-testid="accordion-expandedContent"
        >
          {expandedContent}
        </div>
      )}
    </div>
  );
}
