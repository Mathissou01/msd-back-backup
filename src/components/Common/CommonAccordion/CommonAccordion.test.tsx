import { fireEvent, render, screen } from "@testing-library/react";
import CommonAccordion from "./CommonAccordion";

const expanderLabel = "Mocked expander label";
const expandedContentTestId = "accordion-expandedContent";

describe("CommonAccordion", () => {
  it("renders", () => {
    const { container } = render(
      <CommonAccordion
        content={<span>Mocked content</span>}
        expandedContent={<span>Mocked expanded content</span>}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it("has working props", () => {
    render(
      <CommonAccordion
        content={<span>Mocked content</span>}
        expandedContent={<span>Mocked expanded content</span>}
        expanderLabel={expanderLabel}
      />,
    );
    let expandedContent = screen.queryByTestId(expandedContentTestId);
    expect(expandedContent).toBeNull();
    const content = screen.getByTestId("accordion-content");
    expect(content).toBeInTheDocument();
    const expander = screen.getByTestId("accordion-expander");
    expect(expander).toBeInTheDocument();
    expect(expander).toHaveTextContent("Mocked expander label");
    fireEvent.click(expander);
    expandedContent = screen.getByTestId(expandedContentTestId);
    expect(expandedContent).toBeInTheDocument();
  });
});
