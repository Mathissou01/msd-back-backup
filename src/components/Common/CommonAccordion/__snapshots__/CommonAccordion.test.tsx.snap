// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonAccordion renders 1`] = `
<div>
  <div
    class="c-CommonAccordion"
  >
    <div
      class="c-CommonAccordion__ContentRow"
    >
      <div
        class="c-CommonAccordion__ContentRow_content"
        data-testid="accordion-content"
      >
        <span>
          Mocked content
        </span>
      </div>
      <div
        class="c-CommonAccordion__ContentRow_expander"
        data-testid="accordion-expander"
      >
        <span
          class="c-CommonAccordion__ContentRow_expander_icon"
        />
      </div>
    </div>
  </div>
</div>
`;
